<a href="https://colab.research.google.com/github/MohHasan1/park-ai/blob/main/ParkAI.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!pip install ultralytics

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# /content/drive/MyDrive/dl_data/parking_dataset/data.yaml

In [ ]:
from ultralytics import YOLO

In [ ]:
model = YOLO('yolov8s.pt')  # Load a pre-trained model

In [7]:
results = model.train(data='/content/drive/MyDrive/dl_data/parking_dataset/data.yaml', epochs=50, imgsz=640)

Ultralytics 8.3.98 🚀 Python-3.11.11 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
engine/trainer: task=detect, mode=train, model=yolov8s.pt, data=/content/drive/MyDrive/dl_data/parking_dataset/data.yaml, epochs=50, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=Tr

100%|██████████| 755k/755k [00:00<00:00, 147MB/s]


Overriding model.yaml nc=80 with nc=2

                   from  n    params  module                                       arguments                     
  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]                 
  1                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  2                  -1  1     29056  ultralytics.nn.modules.block.C2f             [64, 64, 1, True]             
  3                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  4                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  5                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  6                  -1  2    788480  ultralytics.nn.modules.block.C2f             [256, 256, 2, True]           
  7                  -1  1   1180672  ultralytics

100%|██████████| 5.35M/5.35M [00:00<00:00, 293MB/s]


AMP: checks passed ✅


train: Scanning /content/drive/MyDrive/dl_data/parking_dataset/train/labels... 2150 images, 0 backgrounds, 0 corrupt: 100%|██████████| 2150/2150 [51:52<00:00,  1.45s/it]


train: New cache created: /content/drive/MyDrive/dl_data/parking_dataset/train/labels.cache
WARNING ⚠️ Box and segment counts should be equal, but got len(segments) = 7841, len(boxes) = 74199. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))


val: Scanning /content/drive/MyDrive/dl_data/parking_dataset/valid/labels... 619 images, 0 backgrounds, 0 corrupt: 100%|██████████| 619/619 [15:04<00:00,  1.46s/it]


val: New cache created: /content/drive/MyDrive/dl_data/parking_dataset/valid/labels.cache
WARNING ⚠️ Box and segment counts should be equal, but got len(segments) = 2068, len(boxes) = 31686. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.
Plotting labels to runs/detect/train/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.001667, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
Image sizes 640 train, 640 val
Using 2 dataloader workers
Logging results to runs/detect/train
Starting training for 50 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/50      9.69G      1.625      1.326      1.427        201        640: 100%|██████████| 135/135 [00:57<00:00,  2.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:11<00:00,  1.71it/s]

                   all        619      31686      0.809      0.747      0.805      0.456



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       2/50      11.4G      1.454     0.9296      1.308        181        640: 100%|██████████| 135/135 [00:53<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.11it/s]

                   all        619      31686      0.886      0.832      0.876      0.491



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       3/50      11.5G        1.4     0.8766      1.271        234        640: 100%|██████████| 135/135 [00:53<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.95it/s]

                   all        619      31686      0.892      0.845      0.879      0.539



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       4/50      11.5G      1.368     0.8689      1.254        538        640: 100%|██████████| 135/135 [00:52<00:00,  2.56it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.06it/s]

                   all        619      31686      0.915      0.815      0.889      0.525



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       5/50      11.6G       1.33     0.8093      1.231        172        640: 100%|██████████| 135/135 [00:53<00:00,  2.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.96it/s]

                   all        619      31686       0.94      0.846      0.905      0.568



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       6/50      13.4G      1.264      0.764      1.198        494        640: 100%|██████████| 135/135 [00:53<00:00,  2.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.13it/s]

                   all        619      31686      0.921      0.861      0.905      0.552



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       7/50       7.9G      1.242     0.7451      1.191        463        640: 100%|██████████| 135/135 [00:53<00:00,  2.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.93it/s]

                   all        619      31686      0.914       0.88      0.912      0.576



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       8/50      9.62G      1.254     0.7468      1.194        140        640: 100%|██████████| 135/135 [00:53<00:00,  2.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.18it/s]


                   all        619      31686       0.91      0.873      0.906      0.557

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       9/50      9.62G      1.217      0.721      1.181        336        640: 100%|██████████| 135/135 [00:53<00:00,  2.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.95it/s]

                   all        619      31686      0.906      0.871      0.904      0.587



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      10/50      9.62G      1.206     0.7229      1.177        237        640: 100%|██████████| 135/135 [00:53<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:08<00:00,  2.32it/s]

                   all        619      31686      0.923      0.885      0.915      0.576



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      11/50      9.63G      1.168     0.6788      1.148        122        640: 100%|██████████| 135/135 [00:53<00:00,  2.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.98it/s]

                   all        619      31686      0.944      0.884       0.93      0.615



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      12/50      9.66G      1.173     0.6855       1.16         98        640: 100%|██████████| 135/135 [00:52<00:00,  2.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:08<00:00,  2.29it/s]

                   all        619      31686      0.943      0.891      0.924       0.63



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      13/50      9.71G      1.152      0.675      1.143        265        640: 100%|██████████| 135/135 [00:52<00:00,  2.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.96it/s]

                   all        619      31686      0.939      0.887      0.928      0.622



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      14/50      9.73G      1.118     0.6453      1.124        639        640: 100%|██████████| 135/135 [00:52<00:00,  2.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:08<00:00,  2.24it/s]

                   all        619      31686      0.954      0.892      0.932      0.628



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      15/50      9.77G      1.113     0.6436      1.114        135        640: 100%|██████████| 135/135 [00:53<00:00,  2.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.94it/s]

                   all        619      31686      0.932      0.881      0.923      0.602



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      16/50      9.82G      1.119     0.6648      1.131        215        640: 100%|██████████| 135/135 [00:53<00:00,  2.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:08<00:00,  2.23it/s]

                   all        619      31686       0.95       0.89       0.93      0.622



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      17/50      9.84G      1.107     0.6442      1.125        584        640: 100%|██████████| 135/135 [00:53<00:00,  2.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  2.00it/s]

                   all        619      31686      0.945       0.89       0.93      0.623



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      18/50      9.88G       1.08     0.6317      1.111        350        640: 100%|██████████| 135/135 [00:52<00:00,  2.59it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.07it/s]

                   all        619      31686      0.954       0.89      0.934      0.652



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      19/50      9.92G      1.066     0.6139      1.102        217        640: 100%|██████████| 135/135 [00:53<00:00,  2.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.02it/s]

                   all        619      31686       0.95      0.889      0.935       0.66



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      20/50      9.95G      1.054     0.6147      1.098        264        640: 100%|██████████| 135/135 [00:51<00:00,  2.62it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.01it/s]

                   all        619      31686      0.954      0.898      0.939      0.661



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      21/50      9.99G      1.027     0.5895      1.077        300        640: 100%|██████████| 135/135 [00:52<00:00,  2.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.01it/s]


                   all        619      31686      0.957      0.898      0.937      0.655

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      22/50        10G       1.04     0.5921      1.092        326        640: 100%|██████████| 135/135 [00:51<00:00,  2.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.04it/s]

                   all        619      31686      0.946      0.907      0.938      0.667



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      23/50      10.1G      1.007      0.578      1.069        100        640: 100%|██████████| 135/135 [00:53<00:00,  2.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.11it/s]


                   all        619      31686      0.946      0.913      0.939      0.667

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      24/50      10.1G     0.9996     0.5813      1.076        498        640: 100%|██████████| 135/135 [00:51<00:00,  2.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.94it/s]

                   all        619      31686       0.95      0.912       0.94       0.68



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      25/50      10.1G     0.9805     0.5675      1.064        251        640: 100%|██████████| 135/135 [00:53<00:00,  2.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.09it/s]

                   all        619      31686      0.958       0.91      0.941      0.669



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      26/50      10.2G      0.993      0.562      1.065        291        640: 100%|██████████| 135/135 [00:51<00:00,  2.62it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.99it/s]

                   all        619      31686      0.963      0.909      0.943      0.687



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      27/50      10.2G     0.9763     0.5663      1.057        881        640: 100%|██████████| 135/135 [00:53<00:00,  2.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.16it/s]

                   all        619      31686      0.957      0.914      0.942       0.69



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      28/50      10.2G     0.9628     0.5464      1.053        738        640: 100%|██████████| 135/135 [00:51<00:00,  2.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.85it/s]

                   all        619      31686      0.958      0.917      0.945      0.692



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      29/50      10.3G     0.9514     0.5362      1.048        208        640: 100%|██████████| 135/135 [00:54<00:00,  2.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.06it/s]

                   all        619      31686       0.96      0.912       0.94      0.693



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      30/50      10.3G     0.9433     0.5355      1.044        278        640: 100%|██████████| 135/135 [00:52<00:00,  2.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.93it/s]

                   all        619      31686      0.961      0.915      0.945      0.698



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      31/50      10.4G     0.9227     0.5236      1.039        198        640: 100%|██████████| 135/135 [00:53<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.02it/s]

                   all        619      31686      0.961      0.922      0.946      0.697



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      32/50      10.4G     0.9279     0.5227      1.038        177        640: 100%|██████████| 135/135 [00:51<00:00,  2.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.95it/s]

                   all        619      31686      0.963      0.912      0.943      0.704



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      33/50      10.4G     0.9272     0.5138      1.038        598        640: 100%|██████████| 135/135 [00:54<00:00,  2.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.05it/s]

                   all        619      31686      0.966      0.914      0.945      0.708



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      34/50      10.5G     0.9088     0.5166      1.028        475        640: 100%|██████████| 135/135 [00:52<00:00,  2.58it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.93it/s]

                   all        619      31686      0.966      0.911      0.943      0.695



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      35/50      10.5G     0.8981     0.5041      1.017        216        640: 100%|██████████| 135/135 [00:53<00:00,  2.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.05it/s]


                   all        619      31686      0.966      0.916      0.946      0.706

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      36/50      10.5G     0.8938     0.5012      1.017        229        640: 100%|██████████| 135/135 [00:52<00:00,  2.59it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.95it/s]

                   all        619      31686      0.966      0.918      0.948      0.709



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      37/50      10.6G     0.8651     0.4868      1.009        645        640: 100%|██████████| 135/135 [00:53<00:00,  2.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.00it/s]


                   all        619      31686      0.967      0.916      0.945      0.714

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      38/50      10.6G     0.8669     0.4914       1.01        186        640: 100%|██████████| 135/135 [00:52<00:00,  2.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  2.00it/s]

                   all        619      31686      0.962      0.923      0.948      0.722



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      39/50      10.7G     0.8693     0.4909      1.009        556        640: 100%|██████████| 135/135 [00:53<00:00,  2.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.96it/s]

                   all        619      31686      0.963       0.92      0.946      0.714



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      40/50      10.7G     0.8452     0.4765     0.9988        303        640: 100%|██████████| 135/135 [00:51<00:00,  2.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.98it/s]

                   all        619      31686       0.97       0.92      0.948      0.723


Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      41/50      10.7G     0.8302     0.4491      1.001        157        640: 100%|██████████| 135/135 [00:52<00:00,  2.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.09it/s]

                   all        619      31686      0.972       0.92      0.949      0.721



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      42/50      10.8G     0.8168      0.435     0.9972        228        640: 100%|██████████| 135/135 [00:49<00:00,  2.72it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.97it/s]

                   all        619      31686      0.966      0.919      0.946      0.717



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      43/50      10.8G     0.8063     0.4307     0.9959         96        640: 100%|██████████| 135/135 [00:49<00:00,  2.72it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.01it/s]

                   all        619      31686      0.968      0.925      0.948      0.732



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      44/50      10.8G     0.7928     0.4237     0.9841        293        640: 100%|██████████| 135/135 [00:50<00:00,  2.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:08<00:00,  2.30it/s]

                   all        619      31686       0.97      0.925       0.95      0.733



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      45/50      10.9G     0.7725     0.4112     0.9772        456        640: 100%|██████████| 135/135 [00:51<00:00,  2.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.10it/s]

                   all        619      31686      0.971      0.923      0.949      0.736



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      46/50      10.9G     0.7761     0.4201     0.9782         67        640: 100%|██████████| 135/135 [00:49<00:00,  2.73it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  2.00it/s]

                   all        619      31686      0.971      0.926       0.95      0.736



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      47/50      10.9G     0.7555     0.4047     0.9775        223        640: 100%|██████████| 135/135 [00:50<00:00,  2.65it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.09it/s]

                   all        619      31686      0.971      0.926       0.95      0.742



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      48/50        11G     0.7458     0.3984     0.9657        245        640: 100%|██████████| 135/135 [00:51<00:00,  2.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.14it/s]

                   all        619      31686      0.971      0.927       0.95       0.74



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      49/50        11G     0.7421     0.4005     0.9678        141        640: 100%|██████████| 135/135 [00:50<00:00,  2.69it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.03it/s]

                   all        619      31686      0.972      0.927      0.951      0.742



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      50/50        11G     0.7328     0.3898     0.9571         85        640: 100%|██████████| 135/135 [00:49<00:00,  2.74it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:10<00:00,  1.92it/s]

                   all        619      31686      0.972      0.925       0.95      0.744



50 epochs completed in 0.880 hours.
Optimizer stripped from runs/detect/train/weights/last.pt, 22.5MB
Optimizer stripped from runs/detect/train/weights/best.pt, 22.5MB

Validating runs/detect/train/weights/best.pt...
Ultralytics 8.3.98 🚀 Python-3.11.11 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
Model summary (fused): 72 layers, 11,126,358 parameters, 0 gradients, 28.4 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:24<00:00,  1.22s/it]


                   all        619      31686      0.972      0.925       0.95      0.744
                   car        591      16642      0.978      0.918      0.946      0.741
                  free        580      15044      0.965      0.932      0.954      0.748
Speed: 0.2ms preprocess, 4.6ms inference, 0.0ms loss, 3.1ms postprocess per image
Results saved to runs/detect/train


In [10]:
model.export(format='onnx')  # Exports the model in ONNX format


Ultralytics 8.3.98 🚀 Python-3.11.11 torch-2.6.0+cu124 CPU (Intel Xeon 2.00GHz)
Model summary (fused): 72 layers, 11,126,358 parameters, 0 gradients, 28.4 GFLOPs

PyTorch: starting from 'runs/detect/train/weights/best.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 6, 8400) (21.5 MB)
requirements: Ultralytics requirements ['onnx>=1.12.0', 'onnxslim', 'onnxruntime-gpu'] not found, attempting AutoUpdate...
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 16.0/16.0 MB 184.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 144.1/144.1 kB 115.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 280.8/280.8 MB 181.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.0/46.0 kB 165.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 86.8/86.8 kB 186.9 MB/s eta 0:00:00

requirements: AutoUpdate success ✅ 16.1s, installed 3 packages: ['onnx>=1.12.0', 'onnxslim', 'onnxruntime-gpu']
requirements: ⚠️ Restart runtime or rerun command for upd

'runs/detect/train/weights/best.onnx'

# 🧠 YOLOv8 Model Training Summary – Parking Dataset

## ✅ Model Details

- **Model Used:** `yolov8s.pt` (YOLOv8 Small)
- **Task:** Object Detection
- **Classes:** `car`, `free`
- **Epochs:** 50
- **Image Size:** 640x640
- **Framework:** Ultralytics YOLOv8.3.98
- **Dataset Location:** `/content/drive/MyDrive/dl_data/parking_dataset/`

---

## 📊 Final Evaluation Metrics (Epoch 50)

| Metric        | Car        | Free       | Overall    |
|---------------|------------|------------|------------|
| **Precision** | 0.978      | 0.965      | 0.972      |
| **Recall**    | 0.918      | 0.932      | 0.925      |
| **mAP@0.50**  | 0.946      | 0.954      | 0.950      |
| **mAP@0.50:0.95** | 0.741  | 0.748      | 0.744      |

---

## 📉 Loss Summary (Epoch 50)

- **Box Loss:** `0.7328`
- **Class Loss:** `0.3898`
- **DFL Loss:** `0.9571`

---

## ⚠️ Warnings

- Your dataset contains **both box and segment annotations**, but the task is `detect`. YOLOv8 automatically removed the segments.
  - **Fix Suggestion:** Use either a detection-only or segmentation-only dataset for best results.

---

## 🧪 Model Validation

After training, the best model was validated:

```bash
Validating runs/detect/train/weights/best.pt...
